### PR TITLE
Lightbox gallery component

### PR DESCRIPTION
**Pull request by augustaspocius**

### CODE REVIEW

File: src/components/ImageSlider.vue [hunk 1]
```diff
@@ -40,48 +54,31 @@ export default {
 };
 </script>
 <template>
+  <Lightbox :current-slide="currentSlide" :images="images" :is-open="lightboxOpen" @close="lightboxOpen = false"
```
Comment (Kristute) [inline src/components/ImageSlider.vue]: There are some warnings in the console
Screenshot from 2023-04-24 10-48-06

File: src/components/Lightbox.vue [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            <!-- Right button -->
+            <button
+                class="absolute top-1/2 -right-6 transform -translate-y-1/2 bg-white rounded-full w-14 h-14 flex justify-center items-center stroke-black hover:stroke-primary-orange"
+                @click="nextSlide">
+                <svg class="" width="13" height="18">
+                    <path d="m2 1 8 8-8 8" stroke-width="3" fill="none" fill-rule="evenodd" />
+                </svg>
+            </button>
+
+            <button class="absolute -top-8 right-0 w-5 h-5 flex justify-center items-center f" @click="$emit('close')">
```
Comment (Kristute) [inline src/components/Lightbox.vue]: The close button could be a bit bigger, and lightbox background a bit darker by the design

File: src/components/ImageSlider.vue [hunk 3]
```diff
@@ -1,7 +1,12 @@
 <script lang="ts">
+import Lightbox from './Lightbox.vue';
 export default {
-  data(): { currentSlide: number, images: string[] } {
+  components: {
+    Lightbox,
+  },
+  data(): { lightboxOpen: boolean, currentSlide: number, images: string[] } {
```
Comment (Kristute) [inline src/components/ImageSlider.vue]: `images` could be more descriptive, such as `productImages`.

File: src/components/ImageSlider.vue [hunk 4]
```diff
@@ -19,13 +24,22 @@ export default {
     },
   },
   methods: {
+    openLightbox() {
```
Comment (Kristute) [inline src/components/ImageSlider.vue]: functions missing types

File: src/components/Lightbox.vue [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+<script lang="ts">
+export default {
+    data() {
+        return {
+            images: [], // initialize with an empty array
+            currentSlide: 0,
+        };
+    },
+    props: {
+        isOpen: Boolean,
```
Comment (Kristute) [inline src/components/Lightbox.vue]: Could be more descriptive, e.x. `isModalOpen`.

File: src/components/Lightbox.vue [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+
+<template>
+    <div v-if="isOpen" class="fixed inset-0 z-50 flex items-start justify-center bg-black bg-opacity-50 pt-32">
+        <div class="relative">
+            <img v-if="images" :src="images[currentSlide]" class="w-full h-auto max-w-[550px] max-h-[90%] md:rounded-xl"
+                alt="product" />
+
+            <!-- Left button -->
+            <button
```
Comment (Kristute) [inline src/components/Lightbox.vue:52]: Your code missing aria attributes. For example, buttons could have aria attributes to help users with disabilities understand their function.

File: src/components/Lightbox.vue [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            currentSlide: 0,
+        };
+    },
+    props: {
+        isOpen: Boolean,
+        currentSlide: Number,
+        images: Array,
+    },
+    methods: {
+        nextSlide() {
```
Comment (Kristute) [inline src/components/Lightbox.vue]: missing types

File: src/components/Lightbox.vue [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            </button>
+
+            <div
+                class="absolute bottom-[-10] left-1/2 transform -translate-x-1/2 w-full flex justify-center items-center space-x-8 py-8">
+                <div v-for="(image, index) in images" :key="index"
+                    class="relative w-full max-w-[88px] rounded-xl cursor-pointer overflow-hidden"
+                    @click="selectThumbnail(index)">
+                    <img :class="[
+                        'w-full ',
+                        currentSlide === index ? '' : ''
```
Comment (Kristute) [inline src/components/Lightbox.vue:84]: You can use a computed property to calculate the `currentImage` based on the `currentSlide` index.

File: src/components/ImageSlider.vue [hunk 9]
```diff
@@ -40,48 +54,31 @@ export default {
...
     </div>
 
     <!-- Left button -->
-    <button
-      class="absolute top-1/2 left-4 transform -translate-y-1/2 bg-white rounded-full w-10 h-10 flex justify-center items-center
-             md:hidden"
-      @click="previousSlide"
-    >
+    <button class="absolute top-1/2 left-4 transform -translate-y-1/2 bg-white rounded-full w-10 h-10 flex justify-center items-center
+                 md:hidden" @click="previousSlide">
```
Comment (Kristute) [inline src/components/ImageSlider.vue]: On the mobile part of the buttons hides outside the frame

**Review by Kristute — CHANGES_REQUESTED** [review]

File: src/components/Lightbox.vue (hunk 2 above)
Comment (augustaspocius) [reply src/components/Lightbox.vue]: Cannot make the button bigger, since svg won't let me do that for some reason

Comment (Kristute) [reply src/components/Lightbox.vue]: You could try using viewbox.
https://css-tricks.com/scale-svg/

**Comment by Kristute** [comment]
Lightbox not working on mobile?

File: src/components/Lightbox.vue (hunk 6 above)
Comment (Kristute) [reply src/components/Lightbox.vue:52]: close button could also have aria

**Review by Kristute — CHANGES_REQUESTED** [review]
Check light-box for mobile

**Pull request merged**